ql/src/java/org/apache/hadoop/hive/ql/ddl/table/AbstractAlterTableAnalyzer.java

```diff
@@ -55,6 +57,9 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
       if (command.getType() == HiveParser.TOK_ALTERTABLE_RENAMEPART) {
         partitionSpec = getPartSpec(partitionSpecNode);
       } else {
+        if (command.getType() == HiveParser.TOK_ALTERTABLE_COMPACT) {
+          HiveConf.setVar(conf, HiveConf.ConfVars.REWRITE_POLICY, Context.RewritePolicy.PARTITION.name());
```

Comment:
Is it applied for partition compaction case or is it added generically to all cases?

Reply:
`rewritePolicy = ALL_PARTITIONS` is added in case of full table compaction for partitioned and unpartitioned tables.

`HiveConf.setVar(conf, HiveConf.ConfVars.REWRITE_POLICY, Context.RewritePolicy.PARTITION.name());`
Is added for partition compaction case only. This code branch is reachable only when `partitionSpecNode != null` and `command.getType() == HiveParser.TOK_ALTERTABLE_COMPACT`